lib/driver.dart

```diff
@@ -0,0 +1,407 @@
+    () async {
+      final Recorder recorder = recorderFactory();
+      final Runner runner = recorder.isTracingEnabled && !_client.isInManualMode
+          ? Runner(
+              recorder: recorder,
+              setUpAllDidRun: () => _client.startPerformanceTracing(benchmarkName),
+              tearDownAllWillRun: _client.stopPerformanceTracing,
+            )
+          : Runner(recorder: recorder);
+
+      final Profile profile = await runner.run();
+      if (!_client.isInManualMode) {
+        await _client.sendProfileData(profile);
+      } else {
+        _printResultsToScreen(profile);
+        print(profile);
+      }
+    },
+    zoneSpecification: ZoneSpecification(
+      print: (Zone self, ZoneDelegate parent, Zone zone, String line) async {
+        if (_client.isInManualMode) {
+          parent.print(zone, '[$benchmarkName] $line');
+        } else {
+          await _client.printToConsole(line);
+        }
+      },
+      handleUncaughtError: (
+        Zone self,
+        ZoneDelegate parent,
+        Zone zone, Object error,
+        StackTrace stackTrace,
+      ) async {
+        if (_client.isInManualMode) {
+          parent.print(zone, '[$benchmarkName] $error, $stackTrace');
+          parent.handleUncaughtError(zone, error, stackTrace);
+        } else {
+          await _client.reportError(error, stackTrace);
+        }
+      },
+    ),
+  );
+}
+
+void _fallbackToManual(String error) {
+  html.document.body.appendHtml('''
+    <div id="manual-panel">
+      <h3>$error</h3>
+
+      <p>Choose one of the following benchmarks:</p>
+
+      <!-- Absolutely position it so it receives the clicks and not the glasspane -->
+      <ul style="position: absolute">
+        ${
+          _benchmarks.keys
+            .map((String name) => '<li><button id="$name">$name</button></li>')
+            .join('\n')
+        }
+      </ul>
+    </div>
+  ''', validator: html.NodeValidatorBuilder()..allowHtml5()..allowInlineStyles());
+
+  for (final String benchmarkName in _benchmarks.keys) {
+    final html.Element button = html.document.querySelector('#$benchmarkName');
+    button.addEventListener('click', (_) {
+      final html.Element manualPanel = html.document.querySelector('#manual-panel');
+      manualPanel?.remove();
+      _runBenchmark(benchmarkName);
+    });
+  }
+}
+
+/// Visualizes results on the Web page for manual inspection.
+void _printResultsToScreen(Profile profile) {
+  html.document.body.remove();
+  html.document.body = html.BodyElement();
+  html.document.body.appendHtml('<h2>${profile.name}</h2>');
+
+  profile.scoreData.forEach((String scoreKey, Timeseries timeseries) {
+    html.document.body.appendHtml('<h2>$scoreKey</h2>');
+    html.document.body.appendHtml('<pre>${timeseries.computeStats()}</pre>');
+    html.document.body.append(TimeseriesVisualization(timeseries).render());
+  });
+}
+
+/// Draws timeseries data and statistics on a canvas.
+class TimeseriesVisualization {
```

Comment:
Any advantages to have the visualization in this package over keeping it in flutter/flutter?

Reply:
This way we can run `flutter run -d chrome -t ... --profile` in the Gallery and see the results in the browser.